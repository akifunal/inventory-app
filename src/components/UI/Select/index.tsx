import React from 'react';

type SelectProps = {
	value?: string;
	label?: string;
	onChange?: (e: React.ChangeEvent<HTMLSelectElement>) => void;
	onMouseEnter?: () => void;
	onMouseLeave?: () => void;
	options?: string[];
};

const Select = ({
	options,
	label,
	value,
	onChange,
	onMouseEnter,
	onMouseLeave,
}: SelectProps) => {
	return (
		<div>
			<label
				htmlFor='countries'
				className='mb-2 block text-sm font-medium text-gray-900 dark:text-gray-400'
			>
				{label}
			</label>
			<select
				id='countries'
				value={value}
				onChange={onChange}
				onMouseEnter={onMouseEnter}
				onMouseLeave={onMouseLeave}
				className='block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500'
			>
				{options?.map((value) => (
					<option value={value} key={value}>
						{value}
					</option>
				))}
			</select>
		</div>
	);
};

export default Select;
