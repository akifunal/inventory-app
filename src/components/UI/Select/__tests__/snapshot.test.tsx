import { render } from '@testing-library/react';
import Select from '../index';

describe('Input component', () => {
	it('should render and match the snapshot with default props', () => {
		// Arrange and Act
		const { asFragment } = render(<Select />);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render and match the snapshot with the correct props', () => {
		// Arrange and Act
		const options = ['Option 1', 'Option 2', 'Option 3'];

		const { asFragment } = render(
			<Select options={options} label='Test label' value={options[1]} />
		);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});
});
