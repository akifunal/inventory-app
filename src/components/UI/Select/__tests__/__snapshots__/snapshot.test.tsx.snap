// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render and match the snapshot with default props 1`] = `
<DocumentFragment>
  <div>
    <label
      class="mb-2 block text-sm font-medium text-gray-900 dark:text-gray-400"
      for="countries"
    />
    <select
      class="block w-full appearance-none rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500"
      id="countries"
    />
  </div>
</DocumentFragment>
`;

exports[`Input component should render and match the snapshot with the correct props 1`] = `
<DocumentFragment>
  <div>
    <label
      class="mb-2 block text-sm font-medium text-gray-900 dark:text-gray-400"
      for="countries"
    >
      Test label
    </label>
    <select
      class="block w-full appearance-none rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500"
      id="countries"
    >
      <option
        value="Option 1"
      >
        Option 1
      </option>
      <option
        value="Option 2"
      >
        Option 2
      </option>
      <option
        value="Option 3"
      >
        Option 3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
